src/System.ComponentModel.TypeConverter/tests/Drawing/ColorConverterTests.cs

```diff
@@ -0,0 +1,446 @@
+using System.Collections.Generic;
+using System.Drawing;
+using System.Globalization;
+using System.Linq;
+using Xunit;
+
+namespace System.ComponentModel.TypeConverterTests
+{
+    public class ColorConverterTests
+    {
+        [Theory]
+        [InlineData(typeof(string))]
+        public void CanConvertFromTrue(Type type)
+        {
+            var conv = new ColorConverter();
+            Assert.True(conv.CanConvertFrom(type));
+            Assert.True(conv.CanConvertFrom(null, type));
+        }
+
+        [Theory]
+        [InlineData(typeof(Rectangle))]
+        [InlineData(typeof(RectangleF))]
+        [InlineData(typeof(Point))]
+        [InlineData(typeof(PointF))]
+        [InlineData(typeof(Color))]
+        [InlineData(typeof(Size))]
+        [InlineData(typeof(SizeF))]
+        [InlineData(typeof(object))]
+        [InlineData(typeof(int))]
+        public void CanConvertFromFalse(Type type)
+        {
+            var conv = new ColorConverter();
+            Assert.False(conv.CanConvertFrom(type));
+            Assert.False(conv.CanConvertFrom(null, type));
+        }
+
+        [Theory]
+        [InlineData(typeof(string))]
+        public void CanConvertToTrue(Type type)
+        {
+            var conv = new ColorConverter();
+            Assert.True(conv.CanConvertTo(type));
+            Assert.True(conv.CanConvertTo(null, type));
+        }
+
+        [Theory]
+        [InlineData(typeof(Rectangle))]
+        [InlineData(typeof(RectangleF))]
+        [InlineData(typeof(Point))]
+        [InlineData(typeof(PointF))]
+        [InlineData(typeof(Color))]
+        [InlineData(typeof(Size))]
+        [InlineData(typeof(SizeF))]
+        [InlineData(typeof(object))]
+        [InlineData(typeof(int))]
+        public void CanConvertToFalse(Type type)
+        {
+            var conv = new ColorConverter();
+            Assert.False(conv.CanConvertTo(type));
+            Assert.False(conv.CanConvertTo(null, type));
+        }
+
+        public static IEnumerable<object[]> ColorData
+        {
+            get
+            {
+                for (int a = 0; a < 256; a += 23)
+                {
+                    for (int r = 0; r < 256; r += 29)
+                    {
+                        for (int g = 0; g < 256; g += 31)
+                        {
+                            for (int b = 0; b < 256; b += 37)
```

Comment:
This scared me for a minute until I noticed that all of these loops were incrementing by something significantly larger than 1.  I thought at first this was creating 4 billion rows for the theory :smile:

That said, this is still creating 6800 rows.  Is that really necessary?  This ColorData is then being used with 8 theories, which means this is contributing roughly 55,000 tests to the suite.